src/Adapter/MSTest.TestAdapter/Extensions/TestCaseExtensions.cs

```diff
@@ -76,15 +77,16 @@ internal static UnitTestElement ToUnitTestElement(this TestCase testCase, string
         var isAsync = (testCase.GetPropertyValue(Constants.AsyncTestProperty) as bool?) ?? false;
         var testClassName = testCase.GetPropertyValue(Constants.TestClassNameProperty) as string;
         var name = testCase.GetTestName(testClassName);
+        var testIdGenerationStrategy = (TestIdGenerationStrategy)testCase.GetPropertyValue(Constants.TestIdGenerationStrategyProperty, (int)TestIdGenerationStrategy.DisplayName);
```

Comment:
Why are you not retrieving as enum directly? `GetPropertyValue<TestIdGenerationStrategy>`?

Reply:
That type doesn't exist in TP it might cause problems when deserializing the attribute, also adds unnecessary metadata gloat.